### PR TITLE
feat(dashboard): add properties section to setting panel

### PR DESCRIPTION
**Pull request by square-li**
## Overview
https://github.com/awslabs/iot-app-kit/issues/462

Redux actions
- add new redux action UpdateAssetQuery
  - this action will assign `propertyId` as the default `refId` if `refId` is not provided, which is useful when applying `styleSettings` to widget.
- add new redux action UpdateAssetDescription to store DescribeAsset results
- add a side effect to UpdateAssetQuery, which will call IotSitewiseClient to describe unknown assets and then store the response by UpdateAssetDescription

UI
- add property section
- add property component
- change property color by color picker
- display property name/alias/dataType/unit when property is found in AssetDescription
- add MessageOverride for i18n

### Scene Composer
For `scene-composer` package changes specifically, you can preview the component in the published storybook artifact. To do this, wait for the `Publish Storybook` action to complete below.

- Click on the workflow details
- Select the Summary item on the left
- Download the zip file

To run the storybook build locally, you need a local static web server:

```
npm install -g httpserver
cd <Extracted Zip Directory>
httpserver
```

Then open the website http://localhost:8080 to run the doc site.

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/dashboard/src/components/sidePanel/sections/propertiesAlarmSection/index.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import React, { FC } from 'react';
```
Comment (square-li) [inline packages/dashboard/src/components/sidePanel/sections/propertiesAlarmSection/index.tsx:1]: This file was `propertiesAlarmSection.tsx`. I moved it here with a new component `PropertyComponent` and renamed it.

File: packages/dashboard/src/components/sidePanel/index.css [hunk 2]
```diff
@@ -2,4 +2,5 @@
   height: 100%;
   max-height: 100vh;
   overflow-y: scroll;
+  min-width: 400px;
```
Comment (mpopv) [inline packages/dashboard/src/components/sidePanel/index.css:5]: Does this cause any issues with the resizable panes? What happens if you resize the side panel down to <400px?

File: packages/dashboard/src/components/sidePanel/sections/thresholdsSection.tsx [hunk 3]
```diff
@@ -55,14 +55,9 @@ const ThresholdComponent: FC<{ path: string; deleteSelf: () => void }> = ({ path
       <div className="threshold-container">
         <SpaceBetween size="xs" direction={'horizontal'}>
           if
-          <Select
-            className="threshold-content-select-label"
-            options={ComparisonOperatorOptions}
-            selectedOption={selectedOption}
-            onChange={onUpdateComparator}
-          />
+          <Select options={ComparisonOperatorOptions} selectedOption={selectedOption} onChange={onUpdateComparator} />
```
Comment (square-li) [inline packages/dashboard/src/components/sidePanel/sections/thresholdsSection.tsx:58]: Removed unused className and then reformatted by prettier. No actual changes.

File: packages/dashboard/src/store/actions/updateAssetQuery/index.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  };
+};
+
+export const updateAssetQuery = (state: DashboardState, action: UpdateAssetQueryAction): DashboardState => {
+  const findAndUpdateWidgetById = (widgets: Widget[]) =>
+    widgets.map((w) => {
+      if (w.id === action.payload.widget.id) {
+        return {
+          ...action.payload.widget,
+          assets: action.payload.assetQuery,
```
Comment (mpopv) [inline packages/dashboard/src/store/actions/updateAssetQuery/index.ts:55]: Will this delete asset(s) previously attached to the widget and replace them with new asset(s) from the payload? Do we always want to do that, or are there circumstances in which we might want to add assets to a widget with some assets already? I'm thinking particularly of dragging multiple assets/properties from the resource explorer onto a single chart, do we want to 'add' each to the chart, or replace what's already there?

Comment (mpopv) [reply packages/dashboard/src/store/actions/updateAssetQuery/index.ts:55]: (Not necessarily saying the way you're doing it here is wrong, totally fine if this is a first step and that's a feature to be added later)

File: packages/dashboard/src/components/sidePanel/index.css (hunk 2 above)
Comment (square-li) [reply packages/dashboard/src/components/sidePanel/index.css:5]: It doesn't cause any issue with resizable panes. When it <400px, setting panel will change to vertically scrollable. This was mainly for limiting the setting panel becoming too narrow to pushing contents around.

File: packages/dashboard/src/store/actions/updateAssetQuery/index.ts (hunk 4 above)
Comment (square-li) [reply packages/dashboard/src/store/actions/updateAssetQuery/index.ts:55]: UpdateAssetQuery simply replaces widget's assets list with the new payload. We might want to have 2 extra actions like RemoveAssetQuery and AddAssetQuery to handle removing and adding assets. Or we need to handle them when creating the payload.

**Review by mpopv — APPROVED** [review]

**Pull request merged**